### PR TITLE
test: add an integration test for kinesis and s3 source

### LINKED ISSUE #11264: add integration test for AWS s3 and kinesis source

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Add an intergration test for kinesis and s3 source. It's identical with `ad-ctr` demo except the data source is from AWS Kinesis and S3, which is simulated with [localstack](https://localstack.cloud/).

resolve #11264

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: integration_tests/kinesis-s3-source/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      - /bin/sh
+      - -c
+      - |
+        export AWS_ACCESS_KEY_ID="test"
+        export AWS_SECRET_ACCESS_KEY="test"
+        export AWS_DEFAULT_REGION="us-east-1" 
+        aws --endpoint-url=http://localstack:4566 kinesis create-stream --stream-name ad-impression
+        aws --endpoint-url=http://localstack:4566 s3api create-bucket --bucket ad-click
+        /datagen --mode ad-ctr --topic ad_impression --qps 10 kinesis --region us-east-1 --name ad-impression --endpoint http://localstack:4566 &
+        /datagen --mode ad-ctr --topic ad_click --qps 10 s3 --region us-east-1 --bucket ad-click --endpoint http://localstack:4566
```
Comment (kwannoel) [inline integration_tests/kinesis-s3-source/docker-compose.yml:59]: Let see if I got this:
1. The docker service `localstack` will support some endpoints. These are mock aws services.
2. We will use `aws-cli` to setup the bucket meta-data locally, so we can use it in our integration test.
3. We run the existing `datagen` binary to sink data into the mock services.

**Comment by kwannoel** [comment]
LGTM, I was looking for something similar for kinesis perf testing locally previously. Glad to see you found a low-overhead solution for it.

**Review by kwannoel — APPROVED** [review]

File: integration_tests/kinesis-s3-source/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    depends_on: [localstack]
+    command:
+      - /bin/sh
+      - -c
+      - |
+        export AWS_ACCESS_KEY_ID="test"
+        export AWS_SECRET_ACCESS_KEY="test"
+        export AWS_DEFAULT_REGION="us-east-1" 
+        aws --endpoint-url=http://localstack:4566 kinesis create-stream --stream-name ad-impression
+        aws --endpoint-url=http://localstack:4566 s3api create-bucket --bucket ad-click
```
Comment (tabVersion) [inline integration_tests/kinesis-s3-source/docker-compose.yml:57]: maybe `awslocal` according to https://docs.localstack.cloud/user-guide/aws/kinesis/

**Review by tabVersion — APPROVED** [review]
quite impressive localstack supports kinesis. 
thanks for the CI, we really need it. 👍

File: integration_tests/kinesis-s3-source/docker-compose.yml (hunk 1 above)
Comment (fuyufjh) [reply integration_tests/kinesis-s3-source/docker-compose.yml:59]: Correct

**Comment by tabVersion** [comment]
resolve #11264

**Comment by fuyufjh** [comment]
Blocked on #11324

**Comment by fuyufjh** [comment]
Test passed: https://buildkite.com/risingwavelabs/integration-tests/builds/216#_

File: integration_tests/kinesis-s3-source/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  datagen:
+    build: ../datagen
+    depends_on: [localstack]
+    command:
+      - /bin/sh
+      - -c
+      - |
+        export AWS_ACCESS_KEY_ID="test"
+        export AWS_SECRET_ACCESS_KEY="test"
+        export AWS_DEFAULT_REGION="us-east-1"
```
Comment (MrCroxx) [inline integration_tests/kinesis-s3-source/docker-compose.yml:55]: This line contains a tailing space but `risedev check` passed. 🤔

**Pull request merged**